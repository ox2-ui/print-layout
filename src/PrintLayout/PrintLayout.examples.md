### PrintLayout

    const fonts = [
      { importName: 'Baloo+Bhaina', family: 'Baloo Bhaina'},
      { importName: 'Pacifico', family: 'Pacifico'},
      { importName: 'Amatic+SC', family: 'Amatic SC'},
    ];

    <PrintLayout
      bodyFont={'Amatic SC'}
      fonts={fonts}
    >
      <h1 style={{fontFamily: 'Baloo Bhaina'}}>Baloo Bhaina - Mist enveloped the ship three hours out from port.</h1>
      <h1 style={{fontFamily: 'Pacifico'}}>Pacifico - Mist enveloped the ship three hours out from port.</h1>
      <h1>Body font - Amatic SC - Mist enveloped the ship three hours out from port.</h1>
    </PrintLayout>
