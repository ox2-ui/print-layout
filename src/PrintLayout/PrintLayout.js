import React, { PropTypes } from 'react';

const styles = {
  wrapper: {
    height: '100%',
  },
};

/**
 * PrintLayout Component
 */
const PrintLayout = ({ children, className, style, fonts, bodyFont }) => {
  const mergedStyles = {
    ...styles.wrapper,
    ...style,
  };

  return (
    <div className={className} style={mergedStyles}>
      <style>{'html, body, #root, #app, #react-root { height: 100%; margin: 0; padding: 0; }'}</style>
      <style>{'::-webkit-scrollbar { display: none; }'}</style>
      {fonts.map((font, index) =>
        <style key={index}>{`@import 'https://fonts.googleapis.com/css?family=${font.importName}';`}</style>
      )}
      <style>{`body { font-family: ${bodyFont}, sans-serif; }`}</style>
      {children}
    </div>
  );
};


PrintLayout.propTypes = {
  /**
   * Font family name to set on the body tag
   */
  bodyFont: PropTypes.string,
  /**
   * Layout content
   */
  children: PropTypes.node.isRequired,
  /**
   * The css class name of the root element.
   */
  className: PropTypes.string,
  /**
   * List of fonts to import
   */
  fonts: PropTypes.array,
  /**
   * Override the inline-styles of the root element.
   */
  style: PropTypes.object,
};

export default PrintLayout;
