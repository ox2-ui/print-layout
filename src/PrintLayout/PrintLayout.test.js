/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import PrintLayout from './PrintLayout';

const fonts = [
  { importName: 'Baloo+Bhaina', family: 'Baloo Bhaina'},
  { importName: 'Pacifico', family: 'Pacifico'},
  { importName: 'Amatic+SC', family: 'Amatic SC'},
];

it('renders correctly', () => {
  const wrapper = mount(
    <PrintLayout
      bodyFont={'Amatic SC'}
      fonts={fonts}
    >
      <h1 style={{fontFamily: 'Baloo Bhaina'}}>Baloo Bhaina - Mist enveloped the ship three hours out from port.</h1>
      <h1 style={{fontFamily: 'Pacifico'}}>Pacifico - Mist enveloped the ship three hours out from port.</h1>
      <h1>Body font - Amatic SC - Mist enveloped the ship three hours out from port.</h1>
    </PrintLayout>
  );

  expect(wrapper).toMatchSnapshot();
});
