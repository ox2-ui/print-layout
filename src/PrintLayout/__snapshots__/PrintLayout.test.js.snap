// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<PrintLayout
  bodyFont="Amatic SC"
  fonts={
    Array [
      Object {
        "family": "Baloo Bhaina",
        "importName": "Baloo+Bhaina",
      },
      Object {
        "family": "Pacifico",
        "importName": "Pacifico",
      },
      Object {
        "family": "Amatic SC",
        "importName": "Amatic+SC",
      },
    ]
  }
>
  <div
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <style>
      html, body, #root, #app, #react-root { height: 100%; margin: 0; padding: 0; }
    </style>
    <style>
      ::-webkit-scrollbar { display: none; }
    </style>
    <style>
      @import 'https://fonts.googleapis.com/css?family=Baloo+Bhaina';
    </style>
    <style>
      @import 'https://fonts.googleapis.com/css?family=Pacifico';
    </style>
    <style>
      @import 'https://fonts.googleapis.com/css?family=Amatic+SC';
    </style>
    <style>
      body { font-family: Amatic SC, sans-serif; }
    </style>
    <h1
      style={
        Object {
          "fontFamily": "Baloo Bhaina",
        }
      }
    >
      Baloo Bhaina - Mist enveloped the ship three hours out from port.
    </h1>
    <h1
      style={
        Object {
          "fontFamily": "Pacifico",
        }
      }
    >
      Pacifico - Mist enveloped the ship three hours out from port.
    </h1>
    <h1>
      Body font - Amatic SC - Mist enveloped the ship three hours out from port.
    </h1>
  </div>
</PrintLayout>
`;
